editor/src/clj/editor/editor_extensions/vm.clj

```diff
@@ -0,0 +1,204 @@
+;; Copyright 2020-2024 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.editor-extensions.vm
```

Comment:
This is a Lua VM ns, semantically it's 1 abstraction lower than `rt`, so the abstraction stack goes like this:
- Lua VM (this ns) — about synchronizing the VM access using locking.
- Lua Runtime (`editor.editor-extensions.rt`) — wrap VM and adds support for non-blocking, long running interactions using coroutines
- Editor Extensions (`editor.editor-extensions`) — integrates Lua Runtime with the editor scripts in the project.